packages/spaces/src/Spaces.js

```diff
@@ -56,6 +56,31 @@ export const sanitizeSpaces = spaces => {
 
 export const SpacesContext = createContext();
 
+const INITIAL_STATE = {
+  spaces: [],
+  loading: true,
+  error: null,
+};
+
+const actions = {
+  SPACES: (_, { spaces }) => ({
+    spaces: spaces || [],
+    error: null,
+    loading: false,
+  }),
+  ERROR: (state, { error }) => ({
+    ...state,
+    loading: false,
+    error,
+  }),
+  LOADING: (state, { loading }) => ({
+    ...state,
+    loading: loading !== undefined ? loading : !state.loading,
+  }),
+};
+
+const spacesReducer = (state, action) => actions[action.type](state, action);
```

Comment:
The only thing I can think of to be cognizant of is what will happen if `actions[action.type] === undefined`.

Reply:
🤷‍♂  In this case we are safe but yeah I think if it was an app built using the state reducer there would need to be an undefined check.

Reply:
That's what I was thinking as well.